jsonschema/tests/test_cli.py

```diff
@@ -683,6 +684,73 @@ def test_successful_validation_of_just_the_schema_pretty_output(self):
             stderr="",
         )
 
+    def test_successful_validate_with_specifying_base_uri_absolute_path(self):
+        absolute_path = os.getcwd()
+        try:
+            schema_file = tempfile.NamedTemporaryFile(
+                mode='w+',
+                prefix='schema',
+                suffix='.json',
+                dir=absolute_path,
+                delete=False
+            )
+            self.addCleanup(os.remove, schema_file.name)
+            schema = """
+                    {"type": "object", "properties": {"KEY1":
+                    {"$ref": %s%s#definitions/schemas"}},
+                    "definitions": {"schemas": {"type": "string"}}}
+                    """ % ("\"", os.path.basename(schema_file.name))
+            schema_file.write(schema)
+        finally:
+            schema_file.close()
+
+        file_prefix = "file:///{}/" if "nt" == os.name else "file://{}/"
+        absolute_path = file_prefix.format(absolute_path)
+        self.assertOutputs(
+            files=dict(some_schema=schema, some_instance='{"KEY1": "1"}'),
+            argv=[
+                "-i", "some_instance",
+                "--base-uri", absolute_path,
+                "some_schema",
+            ],
+            stdout="",
+            stderr="",
+        )
+
+    def test_failure_validate_with_specifying_base_uri_absolute_path(self):
```

Comment:
We should have a test for specifying the wrong base URI as well, and possibly one for an invalid base URI.